File: go-code/introduce_parameter_object/introduce_parameter_object_after.go
package introduce_parameter_object

import "fmt"

type NumberRange struct {
	min int
	max int
}

func (r NumberRange) contains(number int) bool {
	return r.min < number || r.max > number
}

func readingsOutsideRangeAfter(station Station, numberRange NumberRange) {
	for _, r := range station.Readings {
		if !numberRange.contains(r.Temp) {
			fmt.Println("Reading outside range: ", r.Temp)
		}
	}
}

File: java-code/src/main/java/refactoring/basics/introduce_parameter_object/IntroduceParameterObject_After.java
package refactoring.basics.introduce_parameter_object;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class IntroduceParameterObject_After {

  public void readingsOutsideRange(Station station, NumberRange range) {
    station.getReadings().stream()
        .filter(r -> !range.includes(r.getTemp()))
        .forEach(r -> log.info("Reading outside range: " + r.getTemp()));
  }
}



